### PR TITLE
Update license guidelines for projects

### LINKED ISSUE #37: LICENCEs Guideline
During JupyterCon sprints, one of the topic was to update our various repositories to ensure uniforme licensing terms. We got ~50 Pull-Requests about this topic and a few issues appeared or were discovered. This is thus an issue to discuss the licensing Guidelines. 

A) LICENCEs files were added without consideration of Pre-existing COPYING files. 

B) COPYING were renamed to LICENCE without updates of Manifest.in files leading to releases being uploaded without being Properly licensed. 

C) Previous Copyright were removed or about to get removed

D) MIT project were almost relicenced as BSD without authors consents

D') Projects without licenses were added a BSD Licence without all authors approval

E) Some projects (docs, website tutorial) Got BSD and should get CC0? 

F) Copyright Dates we updated to 2017

I'm Going to Focus on F in the following paragraph, but if we get a Guideline it would be good to cover other points. 

According to Brian:

> Here is the US law on copyright notices:

> https://www.copyright.gov/title17/92chap4.html#401

> tl;dr - we should use the first year of publication and don't need to
> put <year>-present as copyrights don't expire.

Jason pointed out : 

> * While the year is the date of the first publication, you may (should?) also list other years that significant contributions were made, as their copyright starts the year those contributions were published.

Fernando

> Quick note - at the sprints in NYC on Saturday we had a brief chat about trying to simplify this as much as possible to avoid all that churn... I suggest we seek the simplest solution that is compliant with law but that avoids all the "January copyright update storm".  

> If simply listing file creation year is legally sufficient, I'm all for that. Like in many other areas, times of significant activity are adequately recorded by the VCS, so I don't think having a bunch of years listed in the file (or an updated endpoint) is really of any use.

And Paul

> I left this as a comment in a recent JupyterLab PR that bumped the copyright verison: we moved to remove the end copyright date from files as we updated them in IPython starting back in 2012. The reason is that it keeps you from having to have a noop commit once per year that updates the whole repo's end dates. For more on this discussion, see ipython/ipython#1644 

So can we get a clear guideline ni this repo and check all the PRs that  have been recently made.

Note: We're asking other project to correctly use our licensing terms when they use some of our code, so let's do things properly.

### PR DESCRIPTION
**Pull request by jasongrout**
This follows up on conversation in https://github.com/jupyter/governance/issues/37 and the [mailing list discussion](https://groups.google.com/forum/#!topic/jupyter/vZpWgw8zKdc) about updating our project license and copyright terms to be more clear and conform better to the opensource.org and spdx BSD license pages.

Fixes #37 

## Summary of updates

* Update primary license name to 3-Clause BSD License to reflect the opensource.org and spdx.org primary names for the license

* Make copyright owner “Project Jupyter Contributors” to reflect discussion in the mailing list thread. We do not view this as a change, but rather as a clarification of our shared copyright model.

* Wrap all lines to 78 or fewer characters

* Update BSD conditions to be enumerated with numbers, following the text on opensource.org and spdx.org

* In condition 3, replace “Jupyter Development Team” with the exact license text “copyright holder”.

* We previously had, in the disclaimer section, a sentence starting with “IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS…”. OWNER is now replaced with HOLDER, to reflect the license text on opensource.org and spdx.org

* Update the per-file license statement with “Project Jupyter Contributors” and “3-Clause BSD License” terminology.

I also edited our copyright policy explanation for grammar and to clarify that for a particular project, its copyright is shared by its contributors (not necessarily *all* Jupyter contributors).

### CODE REVIEW

**Comment by jasongrout** [comment]
According to [governance rules](https://github.com/jupyter/governance/blob/master/governance.md#changing-the-governance-documents), this PR needs approval by 2/3 of the steering council, with at least 80% voting, to be merged. For our current size of 17, that means at least 14 total votes and at least 12 positive votes. Count my vote as 1.

Of course community discussion is welcome as well.

**Comment by jasongrout** [comment]
And of course, comments or suggestions are also encouraged and invited, not just an up-or-down vote.

File: projectlicense.md [hunk 1]
```diff
@@ -1,61 +1,51 @@
...
-their copyright in the commit message of the change, when they commit the
-change to one of the Jupyter repositories.
+Project Jupyter uses a shared copyright model. Each contributor maintains
+copyright over their contributions to Project Jupyter. These contributions are
+typically only changes to the repositories. Thus, Project Jupyter source code,
+in its entirety, is not the copyright of any single person or institution.
+Instead, Project Jupyter source code is the collective copyright of its
+contributors. If individual contributors want to maintain a record of what
+changes or contributions they have specific copyright on, they should indicate
+their copyright in the commit message of the change when they commit the
```
Comment (afshin) [inline projectlicense.md]: I don't have an objection to this language because it specifically says this is to maintain a record. It still remains the case that a copyright holders rights are intact irrespective of whether they explicitly say so.

The language in this clause does make it sound a little bit like those copyright notices *should* be in commit messages, but I think a close reading of this is actually correct, so I am not proposing a change. I am commenting here for clarity and also to invite disagreement if I've misunderstood.

Comment (jasongrout) [reply projectlicense.md]: I had the same thoughts - the existing language was technically fine (though a bit awkward grammatically?), so I hesitated to change it too much.

File: projectlicense.md [hunk 2]
```diff
@@ -1,61 +1,51 @@
...
-This project is licensed under the terms of the Modified BSD License
-(also known as New or Revised or 3-Clause BSD), as follows:
-
-- Copyright (c) 2001-2015, IPython Development Team
-- Copyright (c) 2015-, Jupyter Development Team
+This project is licensed under the terms of the 3-Clause BSD License (also
+known as New or Revised or Modified BSD License, or the BSD-3-Clause license),
+as follows:
 
+Copyright (c) YEAR Project Jupyter Contributors.
```
Comment (ivanov) [inline projectlicense.md]: should `YEAR` here remain a placeholder, or do we put 2015, cribbing from the proposed deletion?

File: projectlicense.md [hunk 3]
```diff
@@ -1,61 +1,51 @@
 # Licensing terms
 
-This project is licensed under the terms of the Modified BSD License
-(also known as New or Revised or 3-Clause BSD), as follows:
-
-- Copyright (c) 2001-2015, IPython Development Team
-- Copyright (c) 2015-, Jupyter Development Team
```
Comment (ivanov) [inline projectlicense.md]: does this deletion retroactively change the copyright that used to be there? I would have thought that any previous BSD license copyrights would remain in place, but maybe it's not a big deal here.

**Comment by ivanov** [comment]
cookies for anyone coming here from the Steering Council email list. 🍪

**Comment by ivanov** [comment]
🍪 🍪 🥛 🍰 🍫 🎁 ☕️

File: projectlicense.md (hunk 2 above)
Comment (jasongrout) [reply projectlicense.md]: Good question. I figured the year would change depending on when that specific repo was started, so I left it a placeholder.

File: projectlicense.md (hunk 3 above)
Comment (jasongrout) [reply projectlicense.md]: I think probably `Jupyter Development Team` can be (optionally?) changed to `Project Jupyter Contributors`. I think any other copyrights probably ought to be left in place, and the `Project Jupyter Contributors` copyright should be appended.

Comment (damianavila) [reply projectlicense.md]: >  I think any other copyrights probably ought to be left in place, and the Project Jupyter Contributors copyright should be appended.

Makes sense to me.

**Comment by jasongrout** [comment]
Update: we've had 9 votes, all positive (as well as a bit of discussion). If you haven't voted, please vote or make a suggestion. I notice that the github approval mechanism does not have a way to vote negative, but we can count the "request changes" option as a negative vote.

**Comment by Ruv7** [comment]
@jasongrout - I will abstain on this vote (I don't know enough about this topic).

**Comment by jasongrout** [comment]
Good point, I hadn't considered abstaining. It seems reasonable that abstaining adjusts the total universe of voting members, so the threshold statement above becomes:

According to governance rules, this PR needs approval by 2/3 of the steering council, with at least 80% voting, to be merged. For our current size of 17 minus one abstaining, that means at least 13 total votes and at least 11 positive votes.

**Comment by ellisonbg** [comment]
I think it does make sense for abstains to reduce the universe size

**Comment by vidartf** [comment]
> According to governance rules, this PR needs approval by 2/3 of the steering council, with at least 80% voting, to be merged. For our current size of 17 minus one abstaining, that means at least 13 total votes and at least 11 positive votes.

Note that you should probably define a deadline for voting, unless this is a first-past-the-post system, as a winning condition (11 of 13 votes) can be made losing (11 of 17 votes) by late voters.

**Comment by jasongrout** [comment]
> Note that you should probably define a deadline for voting, unless this is a first-past-the-post system, as a winning condition (11 of 13 votes) can be made losing (11 of 17 votes) by late voters.

Good point. I'm going by the book here (i.e., the governance docs have an explicit voting threshold for governance repo changes), which does not have a deadline mechanism for votes.

**Comment by jasongrout** [comment]
> > 
> 
> Good point. I'm going by the book here (i.e., the governance docs have an explicit voting threshold for governance repo changes), which does not have a deadline mechanism for votes.

On the other hand, to address your point more directly - perhaps after it passes the quorum threshold of 80%, a reasonable thing to do is to then set a timeframe for any remaining votes before calling the decision.

**Comment by jasongrout** [comment]
Thank you to everyone that responded. We now have quorum (13), with all positive votes, which means that this PR is approved. I'll leave it open until Monday for any remaining dissenting votes or more discussion, then merge Monday if there are no objections.

Thanks again everyone!

**Comment by Carreau** [comment]
Thanks Jason for the ping. 

Approved.

**Comment by rgbkrk** [comment]
Quorum reached, merging.

**Comment by jasongrout** [comment]
Thanks Kyle!

**Comment by jasongrout** [comment]
And thanks again everyone else for participating and adding your thoughts!
